File: client/src/androidTest/java/com/jraska/github/client/TestUITestApp.java
package com.jraska.github.client;

import com.jraska.github.client.http.MockingResponsesHttpComponent;
import com.jraska.github.client.http.HttpComponent;

public class TestUITestApp extends GitHubClientApp {
  @Override protected HttpComponent httpComponent() {
    return MockingResponsesHttpComponent.create();
  }

  @Override protected DaggerAppComponent.Builder componentBuilder() {
    return super.componentBuilder()
      .firebaseModule(new FakeFirebaseModule());
  }
}

File: client/src/androidTest/java/com/jraska/github/client/TestUITestApp.kt
package com.jraska.github.client

import com.jraska.github.client.http.HttpComponent
import com.jraska.github.client.http.ReplayHttpComponent

class TestUITestApp : GitHubClientApp() {
  override fun httpComponent(): HttpComponent {
    return ReplayHttpComponent.create()
  }

  override fun coreComponent(): CoreComponent {
    return FakeCoreComponent()
  }
}
